internal/v1/handler_blueprints.go

```diff
@@ -277,6 +320,78 @@ func (h *Handlers) UpdateBlueprint(ctx echo.Context, blueprintId uuid.UUID) erro
 		return err
 	}
 
+	if !blueprintNameRegex.MatchString(blueprintRequest.Name) {
+		return ctx.JSON(http.StatusUnprocessableEntity, HTTPErrorList{
+			Errors: []HTTPError{{
+				Title:  "Invalid blueprint name",
+				Detail: blueprintInvalidNameDetail,
+			}},
+		})
+	}
+
+	if blueprintRequest.Customizations.Users != nil {
+		for i := range *blueprintRequest.Customizations.Users {
+			requestUser := (*blueprintRequest.Customizations.Users)[i]
+
+			// Make sure every user has either ssh key or password set
+			passwordAndSSHKeyEmptyStringRequest := (requestUser.Password != nil && len(*requestUser.Password) == 0) && (requestUser.SshKey != nil && len(*requestUser.SshKey) == 0)
+			if passwordAndSSHKeyEmptyStringRequest {
+				return ctx.JSON(http.StatusUnprocessableEntity, HTTPErrorList{
+					Errors: []HTTPError{{
+						Title:  "Invalid user",
+						Detail: fmt.Sprintf("User ('%s') must have either a password or an SSH key set.", requestUser.Name),
+					}},
+				})
+			}
```

Comment:
As stated above, this check should be: at least one is present.

Also, I am seeing this code for the 2nd time now, this code should be de-duplicated.